src/main/java/org/opentripplanner/middleware/controllers/api/ApiController.java

```diff
@@ -190,8 +189,14 @@ protected void buildEndpoint(ApiEndpoint baseEndpoint) {
     private ResponseList<T> getMany(Request req, Response res) {
         int limit = HttpUtils.getQueryParamFromRequest(req, LIMIT_PARAM, 0, DEFAULT_LIMIT, 100);
         int offset = HttpUtils.getQueryParamFromRequest(req, OFFSET_PARAM, 0, DEFAULT_OFFSET);
-        Auth0UserProfile requestingUser = getUserFromRequest(req);
-        if (isUserAdmin(requestingUser)) {
+        String userId = HttpUtils.getQueryParamFromRequest(req, USER_ID_PARAM, true);
+        // Filter the response based on the user id, if provided.
+        if (userId != null) {
```

Comment:
A check is needed here whether the requesting user can manage the other user. Without it, an ApiUser could, for instance, poll monitored trips of any OtpUser.

Reply:
Updated.